### DISCUSSION TITLE
Websocket connection failed.  Bridge Server has a major problem

### DISCUSSION THREAD

**Discussion by leo10023** [WalletConnect/walletconnect-monorepo#1100] [Developer Support] [answered]
1.Websocket connection failed, indicating that the wallet does not pop up the authorization prompt after opening the wallet APP. Or sometimes the authorization will pop up, but after clicking "confirm wallet authorization successfully", the WebSocket will not return the connection successfully, and will not return the wallet information,

2.The Websocket connection is slow. After opening the wallet APP and authorization, the wallet information of successful login is returned one minute later,

3.We now judge that Bridge Server has a major problem.

ios APP link

https://apps.apple.com/us/app/area-of-attack/id1613492069

![1](https://user-images.githubusercontent.com/106503870/170912523-632ede39-3ff1-42b0-b642-61b2a4e3162d.jpg)

**Comment by chadyj** [comment]
What is the bridge server URL you are connecting to?
What SDK client and version are you using?

**Reply by leo10023** [reply]
> What is the bridge server URL you are connecting to? What SDK client and version are you using?
bridge server URL  :https://bridge.walletconnect.org
SDK version   ： 1.0

**Reply by pedrouid** [reply]
You are using an old server which is both over capacity and has been deprecated for more than a year now.

I believe Swift packages are not currently using the new default servers which are randomly assigned to alphanumerical subdomains such as "f.bridge.walletconnect.org", "m.bridge.walletconnect.org", "8.bridge.walletconnect.org", etc

In order to solve this you can replicate the Javascript code in this file to randomly assign a server: https://github.com/WalletConnect/walletconnect-monorepo/blob/v1.0/packages/clients/core/src/url.ts

TLDR' randomly select an alphanumerical character to choose a server from your client to propose a session to a wallet

**Reply by leo10023** [reply]
It has been modified according to your method and has not been fixed yet。

There is no problem with our Android version. This happens with the IOS version.

Judging from your experience, where might the problem occur?

Session = new WalletConnectUnitySession(AppData, this, customBridgeUrl, _transport, null, chainId);

https://metamask.app.link/wc?uri=wc%3A7bb93f9a-55a6-4bd9-aed9-43e816d084b2%401%3Fbridge%3Dwss%253A%252F%252F3.bridge.walletconnect.org%26key%3D5035f691d392f72d566f323186d8a47ea1a7bf02c8651b50f1ac708c13cce855

**Reply by leo10023** [reply]
> You are using an old server which is both over capacity and has been deprecated for more than a year now.
> 
> I believe Swift packages are not currently using the new default servers which are randomly assigned to alphanumerical subdomains such as "f.bridge.walletconnect.org", "m.bridge.walletconnect.org", "8.bridge.walletconnect.org", etc
> 
> In order to solve this you can replicate the Javascript code in this file to randomly assign a server: https://github.com/WalletConnect/walletconnect-monorepo/blob/v1.0/packages/clients/core/src/url.ts
> 
> TLDR' randomly select an alphanumerical character to choose a server from your client to propose a session to a wallet

There is a big problem here. Are you still as if nothing had happened?

**Comment by leo10023** [comment, marked as answer]
It has been modified according to your method and has not been fixed yet。

There is no problem with our Android version. This happens with the IOS version.

Judging from your experience, where might the problem occur?

Session = new WalletConnectUnitySession(AppData, this, customBridgeUrl, _transport, null, chainId);

https://metamask.app.link/wc?uri=wc%3A7bb93f9a-55a6-4bd9-aed9-43e816d084b2%401%3Fbridge%3Dwss%253A%252F%252F3.bridge.walletconnect.org%26key%3D5035f691d392f72d566f323186d8a47ea1a7bf02c8651b50f1ac708c13cce855

**Comment by KamilGawelek1** [comment]
I raise topics. I have the same problem with our project. after connecting with the wallet and returning to the application after 1 minute, the application connected with the wallet. There is something wrong with the bridge as 1 month ago it works great. I use random bridge selection and the same v1 after debugging the application, this error shows up after connecting the application to the wallet metamask, if you wait 1 minute or more the application will get the wallet address and the linked event connected will be triggered

```
2022-05-31 15:32:47.170577+0200 appname[3670:1796789] [connection] nw_endpoint_handler_set_adaptive_read_handler [C8.1 [35.241.52.229:443](http://35.241.52.229:443/) ready channel-flow (satisfied (Path is satisfied), viable, interface: en0, ipv4, ipv6, dns)] unregister notification for read_timeout failed

2022-05-31 15:32:47.170759+0200 appname[3670:1796789] [connection] nw_endpoint_handler_set_adaptive_write_handler [C8.1 [35.241.52.229:443](http://35.241.52.229:443/) ready channel-flow (satisfied (Path is satisfied), viable, interface: en0, ipv4, ipv6, dns)] unregister notification for write_timeout failed

Closing Websocket
```

**Reply by leo10023** [reply]
Their response was too slow, which made me very disappointed

**Comment by spatializes** [comment]
Same issue here currently. Wallet connects, then after 1 minute delay, the signing request prompt is triggered. This has broken my dapp. Any possible solutions?

**Comment by KamilGawelek1** [comment]
@drquinn @leo10023 to be clear for you guys not working on walletconnect unity sdk build on ios right?

**Reply by spatializes** [reply]
yes exactly

**Reply by leo10023** [reply]
yes  unity   3D game    Why did you start this company when your reply was so slow?

**Comment by Sushmita1301** [comment]
1.After confirming my wallet authorization in metamask app, the wallet address is not taken by the app. This step fails to authenticate the wallet for the latest 6.5.0 metamask app.
2.It seems to work fine for the older version of metamask app (v6.3.0)
3.I have used all the alphanumerical subdomains of the bridge url such as "f.bridge.walletconnect.org", "m.bridge.walletconnect.org", "8.bridge.walletconnect.org", etc.
It doesn't seem to work on android either for me. Are there any updates or suggestions to where might the problem occur and how I can fix it?
